doc/source/cd/background.md

```diff
@@ -0,0 +1,858 @@
+
+*Figure based on Jay Alammar’s excellent* [visual guide](https://jalammar.github.io/a-visual-guide-to-using-bert-for-the-first-time/) *to the BERT model*
+:::
+
+Alibi Detect contains functionality to leverage pre-trained embeddings
+from HuggingFace’s
+[transformer](https://huggingface.co/transformers/) package. Popular
+models such as [BERT](https://arxiv.org/abs/1810.04805) or
+[DistilBERT](https://arxiv.org/abs/1910.01108) (shown above) can be
+used, but Alibi Detect also allows you to easily use your own embeddings
+of choice. A subsequent dimension reduction step can also be applied if
+necessary, as is done in the [Text drift detection on IMDB movie
+reviews](../examples/cd_text_imdb.nblink) example, where the
+768-dimensional embeddings from the BERT model are passed through an
+untrained AutoEncoder to reduce their dimensionality. Alibi Detect
+allows various types of embeddings to be extracted from transformer
+models, as discussed in
+[alibi_detect.models.tensorflow.embedding](../api/alibi_detect.models.tensorflow.embedding.rst).
+
+#### Graph data
+
+In a similar manner to text data, graph data requires preprocessing
+before drift detection can be performed. This can be done by extracting
+graph embeddings from graph neural network (GNN) encoders, as shown
+below, and demonstrated in the [Drift detection on molecular
+graphs](../examples/cd_mol.nblink) example.
+
+```{image} images/graph_embedding.png
+:align: center
+:alt: A graph embedding
+:width: 550px
+```
+
+### Simple example
+
+For a simple example, we’ll use the [MMD
+detector](methods/mmddrift.ipynb) to check for drift on the
+two-dimensional binary classification problem shown previously. The MMD
+detector is a kernel-based method for multivariate two sample testing.
+Since the number of dimensions is already low, dimension reduction step
+is not necessary here here. For a more advanced example using the [MMD
+detector](methods/mmddrift.ipynb) with dimension reduction, check out
+the [Maximum Mean Discrepancy drift detector on
+CIFAR-10](../examples/cd_mmd_cifar10.nblink) example.
+
+The true model/process is defined as:
+
+$$
+y = \begin{cases}     1 & \text{if } x_2 > sx_1 \\     0 & \text{otherwise} \end{cases}
+$$
+
+where the slope $s$ is set as $s=-1$.
+
+```ipython3
+def true_model(X,slope=-1):
+    z = slope*X[:,0]
+    idx = np.argwhere(X[:,1]>z)
+    y = np.zeros(X.shape[0])
+    y[idx] = 1
+    return y
+```
+
+The 2D input data $\mathbf{x}=(x_1,x_2)$ is generated by sampling
+from a mixture of two Normal distributions:
+
+$$
+\mathbf{x} = \phi_1 \mathcal{N}\left(\left[-1,-1\right]^T, \sigma^2\mathbf{I} \right) + \phi_2 \mathcal{N}\left(\left[1,1\right]^T, \sigma^2\mathbf{I} \right)
+$$
+
+with the standard deviation set at $\sigma=0.8$, and the weights
+set to $\phi_1=\phi_2=0.5$.
+
+```ipython3
+sigma = 0.8
+ref_norm_0 = stats.multivariate_normal([-1,-1], np.eye(2)*sigma**2)
+ref_norm_1 = stats.multivariate_normal([ 1, 1], np.eye(2)*sigma**2)
```

Comment:
Should we try to make the code cells mostly self-contained? E.g. `stats` is a submodule of `scipy` (I assume), but it's not imported. I know some other code cells are example-only which are not even valid Python code, but would be great to have as much executable code as possible. (I assume testing this via execution if it was a Jupyter notebook would be ideal, but I assume we want to keep this as a markdown file?).

Reply:
Added import to cell, and linked to stripped down runnable notebook. We should be able to include the notebook in tests too.